### DISCUSSION TITLE
[General][Java SDK (Component)] TCP KeepAlive And Customized Heartbeat

### DISCUSSION THREAD

**Discussion by QBH-insist** [apache/dubbo#14500] [General]
### Pre-check

- [X] I am sure that all the content I provide is in English.

### Apache Dubbo Component

Java SDK (apache/dubbo)

### Details

If a customized heartbeat protocol is in place, is it necessary to enable TCP keepalive? I reviewed the Dubbo code and noticed
that `NettyClient` had keepalive set to true, but `NettyServer` has is set to false by default. 

I think that a customized heartbeat protocol can check the socket for liveness and availability, is it correct?

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by AlbumenJ** [comment]
Keepalive in L7 is better than L4

**Reply by QBH-insist** [reply]
So, When a customized heartbeat mechanism is in place, it is not necessarily required to set keepalive to true.
